src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -145,7 +145,8 @@ def normalize_disk_info(image_data_disks=None,
     data_disk_sizes_gb = data_disk_sizes_gb or []
     image_data_disks = image_data_disks or []
 
-    data_disk_delete_option = validate_delete_options(attach_data_disks, data_disk_delete_option)
+    if attach_data_disks and data_disk_delete_option:
+        data_disk_delete_option = validate_delete_options(attach_data_disks, data_disk_delete_option)
```

Comment:
Could we put the call of method `_validate_vm_data_disk_delete_option()` here instead of in `_validate_vm_create_storage_profile()`? Such as:
```suggestion
    if  data_disk_delete_option:
        if attach_data_disks:
             data_disk_delete_option = validate_delete_options(attach_data_disks, data_disk_delete_option)
        else:
             data_disk_delete_option = _validate_vm_data_disk_delete_option(attach_data_disks, data_disk_delete_option)
```
In this way, we can concentrate the processing logic of  `data_disk_delete_option` for easy understanding and maintenance